text/entities/0256-entities-data-model.md

```diff
@@ -0,0 +1,789 @@
+    </td>
+    <td><strong>Type</strong>
+    </td>
+    <td><strong>Description</strong>
+    </td>
+   </tr>
+   <tr>
+    <td>Type
+    </td>
+    <td>string
+    </td>
+    <td>Defines the type of the entity. MUST not change during the
+lifetime of the entity. For example: "service" or "host". This field is
+required and MUST not be empty for valid entities.
+    </td>
+   </tr>
+   <tr>
+    <td>Id
+    </td>
+    <td>key/value pair list
+    </td>
+    <td>A set of attributes that identifies the entity.
+<p>
+MUST not change during the lifetime of the entity. The Id must contain
+at least one attribute.
+<p>
+Follows OpenTelemetry <a
+href="https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/common#attribute">common
+attribute definition</a>. SHOULD follow OpenTelemetry <a
+href="https://github.com/open-telemetry/semantic-conventions">semantic
+conventions</a> for attributes.
+    </td>
+   </tr>
+   <tr>
+    <td>Attributes
+    </td>
+    <td>key/value pair list
+    </td>
+    <td>A set of descriptive (non-identifying) attributes of the entity.
+<p>
+MAY change over the lifetime of the entity. MAY be empty. These
+attributes are not part of entity's identity.
+<p>
+“value” follows <a
+href="https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#type-any">any</a>
+value definition in the OpenTelemetry spec - it can be a scalar value,
+byte array, an array or map of values. Arbitrary deep nesting of values
+for arrays and maps is allowed.
+<p>
+SHOULD follow OpenTelemetry <a
+href="https://github.com/open-telemetry/semantic-conventions">semantic
+conventions</a> for attributes.
+    </td>
+   </tr>
+</table>
+
+### Minimally Sufficient Id
+
+Often a number of attributes of an entity are readily available for the telemetry
+producer to compose an Id from. Of the available attributes the entity Id should
+include the minimal set of attributes that is sufficient for uniquely identifying
+that entity. No superfluous attributes should be included in the Id set. For example
+a Process on a host can be uniquely identified by (`process.pid`,`process.start_time`)
+attributes. Adding for example `process.executable.name` attribute to the Id is
+unnecessary and violates the Minimally Sufficient Id rule.
+
+### Examples of Entities
+
+_This section is non-normative and is present only for the purposes of demonstrating
+the data model._
+
+Here are examples of entities, the typical identifying attributes they
+have and some examples of non-identifying attributes that may be
+associated with the entity.
+
+<table>
+   <tr>
+    <td><strong>Entity</strong>
+    </td>
+    <td><strong>Entity Type</strong>
+    </td>
+    <td><strong>Identifying Attributes</strong>
+    </td>
+    <td><strong>Non-identifying Attributes</strong>
+    </td>
+   </tr>
+   <tr>
+    <td>Service
+    </td>
+    <td>"service"
+    </td>
+    <td>service.name (required)
```

Comment:
Maybe instead of "required" we should say "not empty"?
Or even more explicitly `(required, not empty)`.

Same comment applies to other places.

Reply:
The definition in sem conv does not require the value to be non-empty.

Reply:
From [definition](https://opentelemetry.io/docs/specs/semconv/general/attribute-requirement-level/#required):

>All instrumentations MUST populate the attribute.

I think this is good enough for the purposes of this OTEP. This is just an example, it is not a normative language and is not an OTEP topic.

We can dive deeper into this when we work on how we define entity semantic conventions. I added a TODO: https://github.com/orgs/open-telemetry/projects/85/views/1?pane=issue&itemId=63999527

Reply:
Resolving. Feel free to reopen if there are further questions.